src/audio/component.c

```diff
@@ -557,3 +567,15 @@ void audio_stream_copy_to_linear(struct audio_stream *source, int ioffset,
 		snk += bytes_copied;
 	}
 }
+
+/**
+ * @brief frame align limit table, array index is framesize(bytes) - 1 , the value is the
+ * shift index to be aligned with 8 bytes in this framesize case. The max framsize
+ * is SOF_IPC_MAX_CHANNELS * 4.
+ */
+const uint8_t frame_align_8_idx[32] = {
```

Comment:
This is the correct method now, but the data needs to be more structured and per component (since every component will process differently).
e.g.
```c
struct processing_alignment {
    int frame_size;
    int alignment;
};

// in volume.c
#if HIFI3
struct processing_alignment volume_align[] = {
    {4, 8},
    {8,  8},
   //and so on
};
#else   
// and so on
```
This way its really easy to understand and to follow (and debug/fix).